Source/Core/Core/IOS/MIOS.cpp

```diff
@@ -94,9 +102,19 @@ static void ReinitHardware()
   SConfig::GetInstance().m_system = SConfig::System::WiiGC;
   SystemTimers::ChangePPCClock(SystemTimers::Mode::GC);
 
-  // Hardware devices which were initialised in Wii mode need to be reinitialised.
+  // IOS resets the whole hardware (including the PPC) and overwrites mem2.
+  std::memset(Memory::m_pEXRAM, 0, Memory::EXRAM_SIZE);
+  AudioInterface::Reinit();
+  VideoInterface::Preset(DiscIO::IsNTSC(SConfig::GetInstance().m_region));
+  SerialInterface::Reinit();
+  ProcessorInterface::Reinit();
+  // ExpansionInterface::Reinit();
```

Comment:
Having this makes Twilight Princess OSPanic() about not being able to set a VI mode. I haven't figured out why, so I'm leaving this disabled temporarily.